File: security/src/main/java/com/truenine/component/security/autoconfig/DisableSecurityPolicyBean.java
package com.truenine.component.security.autoconfig;

import com.truenine.component.security.annotations.EnableRestSecurity;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
import org.springframework.context.annotation.Bean;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.provisioning.InMemoryUserDetailsManager;
import org.springframework.security.web.SecurityFilterChain;

/**
 * spring security 策略注册器
 *
 * @author TrueNine
 * @since 2022-12-14
 */
@ConditionalOnMissingBean({
  SecurityPolicyBean.class
})
@Slf4j
public class DisableSecurityPolicyBean {

  @Bean
  SecurityFilterChain disableSecurityFilterChain(HttpSecurity security) throws Exception {
    log.warn("警告：生产环境请启用 webSecurity, 使用 {} 来启用并配置 {}",
      EnableRestSecurity.class.getName(),
      SecurityPolicyBean.class.getName());
    return
      security
        .csrf()
        .disable()
        .authorizeHttpRequests()
        .anyRequest()
        .permitAll()
        .and()
        .logout()
        .permitAll()
        .and().build();
  }

  @Bean
  UserDetailsService ssr() {
    return new InMemoryUserDetailsManager();
  }
}

File: security/security-spring/src/main/kotlin/io/github/truenine/composeserver/security/autoconfig/DisableSecurityPolicyBean.kt
package io.github.truenine.composeserver.security.autoconfig

import io.github.truenine.composeserver.security.annotations.EnableRestSecurity
import io.github.truenine.composeserver.slf4j
import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configurers.CsrfConfigurer
import org.springframework.security.config.annotation.web.configurers.LogoutConfigurer
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.provisioning.InMemoryUserDetailsManager
import org.springframework.security.web.SecurityFilterChain

/**
 * Spring Security fallback policy registrar when no SecurityPolicyBean is defined.
 *
 * @author TrueNine
 * @since 2022-12-14
 */
@Configuration
@ConditionalOnMissingBean(SecurityPolicyBean::class)
class DisableSecurityPolicyBean {
  companion object {
    @JvmStatic private val log = slf4j(DisableSecurityPolicyBean::class)
  }

  @Bean
  @Throws(Exception::class)
  fun disableSecurityFilterChain(security: HttpSecurity): SecurityFilterChain {
    log.warn("In production, please enable WebSecurity using {} and configure {}", EnableRestSecurity::class.java.name, SecurityPolicyBean::class.java.name)
    return security
      .csrf { obj: CsrfConfigurer<HttpSecurity> -> obj.disable() }
      .authorizeHttpRequests { a -> a.anyRequest().permitAll() }
      .logout { obj: LogoutConfigurer<HttpSecurity?> -> obj.permitAll() }
      .build()
  }

  @Bean
  fun ssr(): UserDetailsService {
    log.warn("Registering a temporary InMemoryUserDetailsManager")
    return InMemoryUserDetailsManager()
  }
}
